tenseal/cpp/tensors/utils/utils.h

```diff
@@ -10,6 +12,104 @@ namespace tenseal {
 using namespace seal;
 using namespace std;
 
+/**
+ * horizontally scan matrix (vector of vectors)
+ **/
+template <typename T>
+void horizontal_scan(const vector<vector<T>>& src, vector<T>& dst) {
```

Comment:
These operations might be easier to execute using a gsl::multi_span structure. For sure not for this PR, I don't know if they actually help. But they should prevent extra-copies

Reply:
Yeah, I will check this out, maybe they will be helpful.